### PR TITLE
Metadata logic change in index.sh script

### PR DESCRIPTION
**Pull request by vishnuchalla**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

Changing cluster nodes metadata logic to bring it inline with [go-commons](https://github.com/cloud-bulldozer/go-commons/blob/main/ocp-metadata/ocp-metadata.go#L209-L237).

## Related Tickets & Documents

Related slack thread: https://redhat-internal.slack.com/archives/C05CDC19ZKJ/p1698882978201349

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Tested the new logic on my self managed cluster and verified the node counts and types
```
Master Nodes: 3 (Type: m5.4xlarge)
Infra Nodes: 0
Worker Nodes: 3 (Type: m5.4xlarge)
All Nodes: 6
```

### CODE REVIEW

**Comment by jtaleric** [comment]
Tested against a rosa cluster
```    for node in $(oc get nodes --ignore-not-found --no-headers -o custom-columns=:.metadata.name || true); do         labels=$(oc get node "$node" --no-headers -o jsonpath='{.metadata.labels}');         if [[ $labels == *"node-role.kubernetes.io/master"* ]]; then             masters=$((masters + 1));             master_type=$(oc get node "$node" -o jsonpath='{.metadata.labels.beta\.kubernetes\.io/instance-type}');             taints=$(oc get node "$node" -o jsonpath='{.spec.taints}');              if [[ $labels == *"node-role.kubernetes.io/worker"* && $taints == "" ]]; then                 workers=$((workers + 1));             fi;         elif [[ $labels == *"node-role.kubernetes.io/infra"* ]]; then             infra=$((infra + 1));             infra_type=$(oc get node "$node" -o jsonpath='{.metadata.labels.beta\.kubernetes\.io/instance-type}');         elif [[ $labels == *"node-role.kubernetes.io/worker"* ]]; then             workers=$((workers + 1));             worker_type=$(oc get node "$node" -o jsonpath='{.metadata.labels.beta\.kubernetes\.io/instance-type}');         fi;         all=$((all + 1));     done; echo $infra; echo $workers; echo $worker_type; echo $infra_type
3
6
m5.xlarge
r5.xlarge
```

**Review by jtaleric — APPROVED** [review]
lgtm

File: utils/index.sh [hunk 1]
```diff
@@ -41,13 +41,33 @@ setup(){
...
+    worker_type=""
+
+    for node in $(oc get nodes --ignore-not-found --no-headers -o custom-columns=:.metadata.name || true); do
+        labels=$(oc get node "$node" --no-headers -o jsonpath='{.metadata.labels}')
+        if [[ $labels == *"node-role.kubernetes.io/master"* ]]; then
+            masters=$((masters + 1))
+            master_type=$(oc get node "$node" -o jsonpath='{.metadata.labels.beta\.kubernetes\.io/instance-type}')
+            taints=$(oc get node "$node" -o jsonpath='{.spec.taints}')
+
+            if [[ $labels == *"node-role.kubernetes.io/worker"* && $taints == "" ]]; then
```
[the remark this answers is not in the document]
Comment (mukrishn) [reply utils/index.sh:59]: what type of cluster is that ?

[the remark this answers is not in the document]
Comment (vishnuchalla) [reply utils/index.sh:59]: I think it was ROSA if I remember correctly. This check was introduced to fix this [issue](https://github.com/cloud-bulldozer/go-commons/issues/26) which we ran into previously.
cc: @afcollins

[the remark this answers is not in the document]
Comment (jtaleric) [reply utils/index.sh:59]: wouldn't it be SNO?

[the remark this answers is not in the document]
Comment (mukrishn) [reply utils/index.sh:59]: Yes right, probably SNO.

**Review by mukrishn — APPROVED** [review]
lgtm

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**